### PR TITLE
Javadoc: Renamed docker images to be clearer and refresh javadoc-site each OBR main build

### PR DESCRIPTION
**Pull request by jadecarino**
- I have changed all references to 'javadocs' to 'javadoc' as that is the correct term (other than where the directory obr/javadocs is referenced, as cannot change the directory name)
- I have changed the image names and dockerfile names to be clearer which is for the maven repo and which is for the site
- I have created a deployment, ingress and service to host the javadoc site at https://development.galasa.dev/main/javadoc-site which will be refreshed each main OBR build (nothing will be there yet as need to merge this in for the images to be built with the new name and refresh argocd etc)

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: pipelines/pipelines/isolated/build-pr.yaml [hunk 1]
```diff
@@ -1011,7 +1011,7 @@ spec:
         - -Dgalasa.release.repo=file:/workspace/git/$(context.pipelineRun.name)/isolated/mvp/repo
         - -Dgalasa.runtime.repo=https://development.galasa.dev/main/maven-repo/obr
         - -Dgalasa.simplatform.repo=https://development.galasa.dev/main/maven-repo/simplatform
-        - -Dgalasa.javadoc.repo=https://development.galasa.dev/main/maven-repo/javadocs
+        - -Dgalasa.javadoc.repo=https://development.galasa.dev/main/maven-repo/javadoc
         - -Dgalasa.docs.repo=https://nexus.galasa.dev/repository/docs/
         - -Dgalasa.central.repo=https://repo.maven.apache.org/maven2/
         - -Dgalasa.eclipse.repo=https://development.galasa.dev/main/maven-repo/eclipse
```
Comment (techcobweb) [inline pipelines/pipelines/isolated/build-pr.yaml:1017]: Wow. 1000+ lines of one build pipeline. Astounded it's so big.

File: pipelines/pipelines/obr/build-pr.yaml [hunk 2]
```diff
@@ -366,41 +366,41 @@ spec:
...
     - name: context
       value: $(context.pipelineRun.name)/obr/javadocs/docker
     - name: noPush
       value: ""
     - name: dockerfilePath
-      value: automation/dockerfiles/javadocs/javadocs-dockerfile
+      value: automation/dockerfiles/javadoc/javadoc-maven-repo-dockerfile
     - name: buildArgs
       value:
         - "--build-arg=baseVersion=latest"
```
Comment (techcobweb) [inline pipelines/pipelines/obr/build-pr.yaml:406]: These changes all look clearer than what was there. :-)

**Review by techcobweb — APPROVED** [review]
Looks great to me. Like the new names.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**